runtime/onert/frontend/base_loader/include/base_loader.h

```diff
@@ -71,6 +71,8 @@ template <typename LoaderDomain, typename SpecificLoader> class BaseLoader
   {
   }
 
+  virtual std::unique_ptr<ir::Graph> loadSubgraph(const SubGraph *subg) = 0;
```

Comment:
(historical note)
It was intended to use static polymorphism (via SpecificLoader) instead of the dynamic one (via virtual methods).
The workflow suggested was as follows:
- Initially, all the code is contained in BaseLoader.
- When Circle implementation of a specific concept begins to differ from the TFLite counterpart (as happened with SubGraphs and some Circle-specific operators), the related code is completely removed from the base loader and implemented in both derived loaders.
- When there become too many differences between the two loaders, the base loader can be removed, leaving two completely independent loaders (Circle and TFLite).
---
I must admit that the intention is not clear from the code, so the implementation of the idea might not be the best.

Virtual methods can also help to achieve the same goal. Applying virtual methods for such purposes is known as "template method pattern" (although it has nothing to do with C++ templates, it's just a name clash). It is well known pattern, so the code using it should be more clear to developers than the former approach using static polymorphism.

Please note that the virtual function should be `private` in this case (in both base and derived classes). More on this [here](http://www.gotw.ca/publications/mill18.htm).

PS
It is worth noting that each virtual method added to BaseLoader is actually two methods (one for each loader). This is because TFLiteLoader and CircleLoader derive from different base classes, thanks to template parameter of BaseLoader.

Reply:
Thanks for your kindly explanation. I will move the virtual function to be `private`.